tests/base.py

```diff
@@ -148,6 +149,60 @@ def clear_tables(self, table_names=ALL_TABLE_NAMES_TO_CLEAR):
     def random_string_generator(size=6, chars=string.ascii_uppercase + string.digits):
         return ''.join(random.choice(chars) for x in range(size))
 
+    @staticmethod
+    def random_decimal_generator(digits=string.digits):
```

Comment:
We have control over the data here right, so instead of randomizing the value can we hardcode a few different values that cover each test case? Like have a new row for each case.

Reply:
Yes we do, but I'm not sure we know what all the problem test cases are.  Given the existing cases start with 100 rows it seemed good to randomize to look for additional problem cases.

Reply:
It looks like you identified some boundaries in your randomization. I think testing the max min, zero and something in between those boundaries would be sufficient. But I guess there are multiple boundaries to consider. Max/min value, max/min precision, min/max scale 🤯 . The hp-mysql and hp-postgres tests have examples for this.

Reply:
Agreed, I'm also not sure how different formatting is handled and if signed or unsigned has significant impact.  Are you suggesting a refactor to align with the hp-mysql and hp-postgres approaches or are we good to leave it random to flush out all of these over time?  To be fair min precision is not currently covered.

Reply:
I think it should be refactored to hardcode values for the different partitions you're covering. The random approach is a good topic to get @HarrisonMarcRose opinion on. But I think it made sense for field selection with google-ads because their weren't clear partitions in the available fields, where we could confidently say if x works so does y. But there are almost always clear partitions in numeric datatypes. And in most datatypes I think.

Reply:
Refactored to a static list of 20 boundry-ish values that are cycled through by the generator.